cdk/src/open-data-platform/data-plane/data-import/handler-factory.ts

```diff
@@ -48,59 +61,79 @@ export const geoJsonHandlerFactory = (
       ]);
 
       pipeline
-        .on('data', async (batch: any[]) => {
-          console.log(`Processing batch. ${numberRows} rows have been read so far.`);
-          // Pause reads while inserting into db.
-          pipeline.pause();
-          await Promise.all(
-            batch.map(async (row: any) => {
-              // Stop processing if at the limit.
-              if (numberRows >= rowLimit) {
-                console.log('Stopping after row write limit:', rowLimit);
-                pipeline.destroy();
-                return;
-              }
-              // Multiple rows may be processed concurrently, so increment and check the limit
-              // before starting the processing step.
-              ++numberRows;
-              await callback(row, db);
-            }),
-          );
-          pipeline.resume();
+        .on('data', async (rows: any[]) => {
+          // Skip rows up to offset.
+          if (skippedRowCount + batchSize <= rowOffset) {
+            skippedRowCount += batchSize;
+            console.log(
+              `Skipping batch of ${rows.length} rows. ${skippedRowCount}/${rowOffset} offset rows have been skipped so far.`,
+            );
+            return;
+          }
+
+          console.log(`Processing batch of ${rows.length} rows.`);
+          const promise = callback(rows, db);
+          promises.push(promise);
```

Comment:
How come this is pushing to the list and then awaiting it? I think awaiting this would take away the parallelism and really slow down the writes / reads

Reply:
I added a comment describing why.